File: backend/app/schemas/definitions.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class UnitPayload(BaseModel):
    name: str = Field(min_length=1, max_length=50)
    is_active: bool = True


class UnitResponse(BaseModel):
    id: uuid.UUID
    name: str
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class TaxRatePayload(BaseModel):
    label: str = Field(min_length=1, max_length=100)
    rate: float = Field(ge=0, le=100)
    is_active: bool = True


class TaxRateResponse(BaseModel):
    id: uuid.UUID
    label: str
    rate: float
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class PaymentTermPayload(BaseModel):
    label: str = Field(min_length=1, max_length=100)
    days: int = Field(ge=0)
    is_active: bool = True


class PaymentTermResponse(BaseModel):
    id: uuid.UUID
    label: str
    days: int
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class CategoryPayload(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    is_active: bool = True


class CategoryResponse(BaseModel):
    id: uuid.UUID
    name: str
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class BankAccountPayload(BaseModel):
    bank_name: str = Field(min_length=1, max_length=100)
    branch_name: str = Field(min_length=1, max_length=100)
    branch_code: str = Field(min_length=1, max_length=20)
    currency: str = Field(default="TRY", pattern=r"^[A-Z]{3}$")
    account_number: str = Field(min_length=1, max_length=50)
    iban: str = Field(min_length=15, max_length=34)
    is_active: bool = True


class BankAccountResponse(BaseModel):
    id: uuid.UUID
    bank_name: str
    branch_name: str
    branch_code: str
    currency: str
    account_number: str
    iban: str
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class NotePayload(BaseModel):
    label: str = Field(min_length=1, max_length=100)
    content: str = Field(min_length=1, max_length=1000)
    is_active: bool = True


class NoteResponse(BaseModel):
    id: uuid.UUID
    label: str
    content: str
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/definitions.ts
export interface DefinitionUnit {
  id: string
  name: string
  is_active: boolean
  is_default: boolean
  created_at: string
}

export interface UnitPayload {
  name: string
  is_active?: boolean
}

export interface DefinitionTaxRate {
  id: string
  label: string
  rate: number
  is_active: boolean
  is_default: boolean
  created_at: string
}

export interface TaxRatePayload {
  label: string
  rate: number
  is_active?: boolean
}

export interface DefinitionPaymentTerm {
  id: string
  label: string
  days: number
  is_active: boolean
  created_at: string
}

export interface PaymentTermPayload {
  label: string
  days: number
  is_active?: boolean
}

export interface DefinitionCategory {
  id: string
  name: string
  is_active: boolean
  created_at: string
}

export interface CategoryPayload {
  name: string
  is_active?: boolean
}

export interface DefinitionBankAccount {
  id: string
  bank_name: string
  branch_name: string
  branch_code: string
  currency: string
  account_number: string
  iban: string
  is_active: boolean
  created_at: string
}

export interface BankAccountPayload {
  bank_name: string
  branch_name: string
  branch_code: string
  currency: string
  account_number: string
  iban: string
  is_active?: boolean
}

export interface DefinitionNote {
  id: string
  label: string
  content: string
  is_active: boolean
  is_default: boolean
  created_at: string
}

export interface NotePayload {
  label: string
  content: string
  is_active?: boolean
}
